### PR TITLE
ActionCable: Allow adding custom paths for channels

### PR DESCRIPTION
**Pull request by droptheplot**
Created mountable engine for `rails 5.0.0.beta1` and found that channels are loaded only from main application. Since `channels_path` can store only one path, so I change it into array.

### CODE REVIEW

File: actioncable/lib/action_cable/server/configuration.rb [hunk 1]
```diff
@@ -15,13 +15,15 @@ def initialize
...
+        @channels_path = [Rails.root.join('app/channels')]
 
         @disable_request_forgery_protection = false
       end
 
       def channel_paths
-        @channels ||= Dir["#{channels_path}/**/*_channel.rb"]
+        @channels ||= channels_path.collect do |channel_path|
+          Dir["#{channel_path}/**/*_channel.rb"]
+        end.flatten
```
Comment (kaspth) [inline actioncable/lib/action_cable/server/configuration.rb]: How about:

```ruby
channels_path.flat_map { |path| Dir["#{path}/**/*_channel.rb"] }
```

Comment (kaspth) [inline actioncable/lib/action_cable/server/configuration.rb]: While we're here we should change the instance variable to match the method name.

Also having a `channels_path` as an array feels off, when we have `channel_paths`. Don't see an entirely good direction here, but perhaps renaming `channels_path` to `channels_paths` could do?

Comment (droptheplot) [inline actioncable/lib/action_cable/server/configuration.rb]: I would suggest `channel_load_paths` because it would be harder to get confused.

**Comment by rafaelfranca** [comment]
This should be handled by the Railtie. You can merge this one but we'll need to use `Paths` object to do that.

**Comment by kaspth** [comment]
@rafaelfranca I've been looking at `Rails::Paths` but don't see how we can use it and still have Action Cable be used independently of Rails without also adding a dependency on Railties.

**Pull request merged**